packages/mui-utils/src/useTimeout.ts

```diff
@@ -0,0 +1,41 @@
+'use client';
+import useLazyRef from './useLazyRef';
+import useOnMount from './useOnMount';
+
+export class Timeout {
+  static create() {
+    return new Timeout();
+  }
+
+  currentId: number = 0;
+
+  /**
+   * Executes `fn` after `delay`, clearing any previously scheduled call.
+   */
+  start(delay: number, fn: Function) {
```

Comment:
Don't we already have `debounce` for this exact use-case.

Reply:
I'm not sure which function you're talking about, but "debounce" isn't the operation that's being done here. This is just a standard `setTimeout`, with the added logic to avoid `clearTimeout` if not needed.

Reply:
I see what you mean. It does act like "debounce". But this is meant to encapsulate & abstract the logic for one "timeout" object that can only have one active timeout, which is why it needs to cancel previously scheduled calls. The other important part that this hook provides, besides avoiding the `clearTimeout` call if possible, is automated resource management: there is no need to manually add an effect to avoid leaking the `setTimeout`.